kuka_external_control_sdk/kss/src/message_builder.cc

```diff
@@ -82,6 +90,33 @@ void MotionState::CreateFromXML(const char *incoming_xml) {
 
   next_value_idx += endptr - &incoming_xml[next_value_idx];
   next_value_idx += kAttributeSuffix.length();
+
+  // External axes
+  // this -1 is to account for that there is no " before the fist attribute
+  next_value_idx += kExtJointPositionsPrefix.length() - 1;
+
+  for (int i = 0; i < kFixSixAxes; ++i) {
+    std::size_t dbl_length = 0;
+    next_value_idx += std::floor(std::log10(

```

Comment:
TODO might be enough, but as we have separated ext axes, we will not have more than 10 and this log10 magic is not needed

Reply:
Removed.